src/index.js

```diff
@@ -32,3 +32,7 @@ export const deepMap = _.curry((fn, obj, _map = map, is = isTraversable) =>
 // Misc
 // ----
 export const testRegex = regex => regex.test.bind(regex)
+
+// Gets all the regexp.exec indexes and their respective lastIndexes
+export const postings = _.curry((r, s) =>
+    _.map(x => (m => [m.index, r.lastIndex])(r.exec(s)), s.match(r)))
```

Comment:
Two things:
1. The anonymous function there isn't needed
```_.curry((r, s) => _.map(x => (m => [m.index, r.lastIndex])(r.exec(s)), s.match(r)))```
vs
```_.curry((r, s) =>_.map(x => [r.exec(s).index, r.lastIndex], s.match(r)))```

2. Using `match` to get an array of matches to then map over without worrying about the contents is very clever but pretty dirty (and less performant because it has to run the regex multiple times), so we should probably be more up front about the dirtiness with something like `_.times`
``` _.curry((r, s) => _.times(x => [r.exec(s).index, r.lastIndex], s.match(r).length))```